### ISSUE TITLE
[Bug]: plug in show server error after trying to export to CSV

### ISSUE THREAD

**Issue by aru1gj01** [ryanmitchell/ti-ext-reports#23]
### What happened?

When press " Export to CSV " , page show to server error 500

### What did you expect to happen?

I check system log and see : 

ErrorException: Illegal string offset 'key' 
in /home/orderfood/domains/orderfood.nz/public_html/extensions/thoughtco/reports/controllers/Builder.php:183

### Version

2.0.6

### Relevant log output

```shell
ErrorException: Illegal string offset 'key' 
in /home/orderfood/domains/orderfood.nz/public_html/extensions/thoughtco/reports/controllers/Builder.php:183
```

**Comment by ryanmitchell** [comment]
I need the full error trace (APP_DEBUG to true in your .env file) rather than just the last error.

Edit: Can you also tell me what version of the extension you are using as the line numbers dont line up with the latest code

**Comment by iva-system** [comment]
Hi Guys, 
i got that error as well:
I have 2.0.6


"ErrorException: Trying to access array offset on value of type null
in /customers/e/0/d/localhost/httpd.www/eg/extensions/thoughtco/reports/formwidgets/ReportsTable.php:17Stack trace:
#0 /customers/e/0/d/localhost/httpd.www/eg/extensions/thoughtco/reports/formwidgets/ReportsTable.php(17): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'Trying to acces...', '/customers/e/0/...', 17)
#1 [internal function]: Thoughtco\Reports\FormWidgets\ReportsTable->getDataTableRecords('0', '10', '')
#2 /customers/e/0/d/localhost/httpd.www/eg/vendor/tastyigniter/flame/src/Traits/EventEmitter.php(147): call_user_func_array(Array, Array)
#3 /customers/e/0/d/localhost/httpd.www/eg/app/admin/widgets/Table.php(174): Admin\Classes\BaseWidget->fireEvent('table.getRecord...', Array, true)
#4 [internal function]: Admin\Widgets\Table->onGetRecords('view', '2')
#5 /customers/e/0/d/localhost/httpd.www/eg/app/admin/classes/AdminController.php(474): call_user_func_array(Array, Array)
#6 /customers/e/0/d/localhost/httpd.www/eg/app/admin/classes/AdminController.php(344): Admin\Classes\AdminController->runHandler('Reportdatatable...', Array)
#7 /customers/e/0/d/localhost/httpd.www/eg/app/admin/classes/AdminController.php(223): Admin\Classes\AdminController->processHandlers()
#8 /customers/e/0/d/localhost/httpd.www/eg/app/system/classes/Controller.php(115): Admin\Classes\AdminController->remap('view', Array)
#9 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): System\Classes\Controller->runAdmin('thoughtco/repor...')
#10 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\Routing\Controller->callAction('runAdmin', Array)
#11 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Routing/Route.php(262): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(System\Classes\Controller), 'runAdmin')
#12 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\Routing\Route->runController()
#13 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Routing/Router.php(721): Illuminate\Routing\Route->run()
#14 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#15 /customers/e/0/d/localhost/httpd.www/eg/app/admin/middleware/LogUserLastSeen.php(23): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#16 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Admin\Middleware\LogUserLastSeen->handle(Object(Illuminate\Http\Request), Object(Closure))
#17 /customers/e/0/d/localhost/httpd.www/eg/vendor/tastyigniter/flame/src/Translation/Middleware/Localization.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#18 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Igniter\Flame\Translation\Middleware\Localization->handle(Object(Illuminate\Http\Request), Object(Closure))
#19 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#20 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Routing\Middleware\SubstituteBindings->handle(Object(Illuminate\Http\Request), Object(Closure))
#21 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#22 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#23 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#24 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\Session\Middleware\StartSession->handleStatefulRequest(Object(Illuminate\Http\Request), Object(Illuminate\Session\Store), Object(Closure))
#25 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#26 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#27 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#28 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#29 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#30 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#31 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Routing/Router.php(723): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#32 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Routing/Router.php(698): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#33 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Routing/Router.php(662): Illuminate\Routing\Router->runRoute(Object(Illuminate\Http\Request), Object(Illuminate\Routing\Route))
#34 /customers/e/0/d/localhost/httpd.www/eg/vendor/tastyigniter/flame/src/Router/CoreRouter.php(22): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#35 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(167): Igniter\Flame\Router\CoreRouter->dispatch(Object(Illuminate\Http\Request))
#36 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#37 /customers/e/0/d/localhost/httpd.www/eg/vendor/tastyigniter/flame/src/Setting/Middleware/SaveSetting.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#38 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Igniter\Flame\Setting\Middleware\SaveSetting->handle(Object(Illuminate\Http\Request), Object(Closure))
#39 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#40 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
#41 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\TrimStrings->handle(Object(Illuminate\Http\Request), Object(Closure))
#42 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#43 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle(Object(Illuminate\Http\Request), Object(Closure))
#44 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#45 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance->handle(Object(Illuminate\Http\Request), Object(Closure))
#46 /customers/e/0/d/localhost/httpd.www/eg/vendor/fruitcake/laravel-cors/src/HandleCors.php(38): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#47 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fruitcake\Cors\HandleCors->handle(Object(Illuminate\Http\Request), Object(Closure))
#48 /customers/e/0/d/localhost/httpd.www/eg/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#49 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\Proxy\TrustProxies->handle(Object(Illuminate\Http\Request), Object(Closure))
#50 /customers/e/0/d/localhost/httpd.www/eg/extensions/igniter/api/vendor/dingo/api/src/Http/Middleware/Request.php(111): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#51 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Dingo\Api\Http\Middleware\Request->handle(Object(Illuminate\Http\Request), Object(Closure))
#52 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#53 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(142): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#54 /customers/e/0/d/localhost/httpd.www/eg/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(111): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#55 /customers/e/0/d/localhost/httpd.www/eg/index.php(53): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#56 {main}"

**Comment by aru1gj01** [comment]
Here is the full error from debug.
> 
I use extension version : 2.0.6
......

ErrorException
in /home/orderfood/domains/orderfood.nz/public_html/extensions/thoughtco/reports/controllers/Builder.php (line 183)
            $csv_columns = [];            $csv_headings = [];            foreach ($model->csv_columns as $list_col) {                $col = $list_col['column'];                $csv_columns[] = $col['key'];                $csv_headings[] = $list_col['label'];            }            $data = $data->map(function ($row) use ($csv_columns) {                return $row->only($csv_columns);
HandleExceptions->handleError()
in /home/orderfood/domains/orderfood.nz/public_html/extensions/thoughtco/reports/controllers/Builder.php (line 183)
Builder->view()
call_user_func_array()
in /home/orderfood/domains/orderfood.nz/public_html/app/admin/classes/AdminController.php (line 288)
AdminController->execPageAction()
in /home/orderfood/domains/orderfood.nz/public_html/app/admin/classes/AdminController.php (line 228)
AdminController->remap()
in /home/orderfood/domains/orderfood.nz/public_html/app/system/classes/Controller.php (line 115)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Routing/Controller.php -> runAdmin (line 54)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php -> callAction (line 45)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Routing/Route.php -> dispatch (line 262)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Routing/Route.php -> runController (line 205)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php -> run (line 721)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> Illuminate\Routing\{closure} (line 128)
Pipeline->Illuminate\Pipeline\{closure}()
in /home/orderfood/domains/orderfood.nz/public_html/app/admin/middleware/LogUserLastSeen.php (line 23)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/tastyigniter/flame/src/Translation/Middleware/Localization.php -> Illuminate\Pipeline\{closure} (line 27)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php -> Illuminate\Pipeline\{closure} (line 50)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php -> Illuminate\Pipeline\{closure} (line 49)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php -> Illuminate\Pipeline\{closure} (line 121)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php -> handleStatefulRequest (line 64)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php -> Illuminate\Pipeline\{closure} (line 37)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php -> Illuminate\Pipeline\{closure} (line 67)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> Illuminate\Pipeline\{closure} (line 103)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php -> then (line 723)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php -> runRouteWithinStack (line 698)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php -> runRoute (line 662)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/tastyigniter/flame/src/Router/CoreRouter.php -> dispatchToRoute (line 22)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php -> dispatch (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> Illuminate\Foundation\Http\{closure} (line 128)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/tastyigniter/flame/src/Setting/Middleware/SaveSetting.php -> Illuminate\Pipeline\{closure} (line 27)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php -> Illuminate\Pipeline\{closure} (line 21)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php -> handle (line 40)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php -> Illuminate\Pipeline\{closure} (line 27)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php -> Illuminate\Pipeline\{closure} (line 86)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/fruitcake/laravel-cors/src/HandleCors.php -> Illuminate\Pipeline\{closure} (line 38)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/fideloper/proxy/src/TrustProxies.php -> Illuminate\Pipeline\{closure} (line 57)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/extensions/igniter/api/vendor/dingo/api/src/Http/Middleware/Request.php -> Illuminate\Pipeline\{closure} (line 111)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> handle (line 167)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php -> Illuminate\Pipeline\{closure} (line 103)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php -> then (line 142)
in /home/orderfood/domains/orderfood.nz/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php -> sendRequestThroughRouter (line 111)
Kernel->handle()
in /home/orderfood/domains/orderfood.nz/public_html/index.php (line 53)

**Comment by ryanmitchell** [comment]
Fixed in 2.0.8
